### PR TITLE
Flexible responses

### PR DESCRIPTION
**Pull request by jpollock**

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @jpollock,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

**Comment by lightbend-cla-validator** [comment]
Hi @jpollock,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

**Comment by marcellanz** [comment]
Awesome @jpollock ! great to see contributions to the python user language support for Cloudstate.

With changes seen like
`pip install -U akkaserverless`

I wonder what the story is about that PR. Shouldn't it be made on the lightbend/lightbend-labs organization instead of here at the Cloudstate repository? This PR transforms a user language support library for cloudstate into a Akka Serverless (AS) SDK compatible with the AS protocol of some of the later beta releases and is incompatible with this project here.

I think it would be better to fork/import this repository into one of these organizations instead of transforming it here. The Github UI lets you import an existing repository while keeping the git history.

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Contributors
+
+## Special thanks for all the people who had helped this project so far:
```
Comment (marcellanz) [inline CONTRIBUTORS.md:3]: Thanks for keeping a Contributors list while migrating this to AS.

File: Description.md [hunk 2]
```diff
@@ -1,30 +1,21 @@
...
-Cloudstate is a specification, protocol, and reference implementation for providing distributed state management patterns suitable for **Serverless** computing. 
-The current supported and envisioned patterns include:
+Read more [here](https://developer.lightbend.com/docs/akka-serverless/index.html) and sign-up [here](https://console.akkaserverless.com/p/register#) for a free account. 
 
-* **Event Sourcing**
-* **Conflict-Free Replicated Data Types (CRDTs)**
-* **Key-Value storage**
-* **P2P messaging**
-* **CQRS read side projections**
+The Akka Serverless Python user language support is a library that implements the Akka Serverless protocol and offers an pythonistic API
```
Comment (marcellanz) [inline Description.md:5]:
```suggestion
The Akka Serverless Python SDK is a library that implements the Akka Serverless protocol and offers a pythonistic API 
```

User Language Support Library is old Cloudstate speak I think.

Comment (pvlugter) [reply Description.md:5]: Yes, Akka Serverless is using `SDK`.

File: README.md [hunk 3]
```diff
@@ -1,13 +1,19 @@
-# Python User Language Support
-Python User Language Support for [Cloudstate](https://github.com/cloudstateio/cloudstate).
+# Akka Serverless Python SDK
+
+Source code for the Akka Serverless Python package.
+
+For more information see the documentation for [implementing Akka Serverless services in JavaScript](https://developer.lightbend.com/docs/akka-serverless/javascript/).
+
+
+This package is a fork of the original [Python SDK](https://github.com/cloudstateio/python-support) for [Cloudstate](https://cloudstate.io/). This Akka Serverless package is heavily indebted to the [C]loudstate contributors](https://github.com/cloudstateio/python-support/graphs/contributors), especially [Adriano Santos](https://github.com/sleipnir) and [Marcel Lanz](https://github.com/marcellanz).
```
Comment (pvlugter) [inline README.md:8]:
```suggestion
This package is a fork of the original [Python SDK](https://github.com/cloudstateio/python-support) for [Cloudstate](https://cloudstate.io/). This Akka Serverless package is heavily indebted to the [Cloudstate contributors](https://github.com/cloudstateio/python-support/graphs/contributors), especially [Adriano Santos](https://github.com/sleipnir) and [Marcel Lanz](https://github.com/marcellanz).
```

File: akkaserverless/config.4,txt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      manage-topics-and-subscriptions = "by-proxy"
+    }
+  }
+}
+
+
+get roadmap -
+eng discussion 
+
+moving -
```
Comment (pvlugter) [inline akkaserverless/config.4,txt:28]: @jpollock, realise this PR is not being merged here, but this looks accidental. The `config.3.txt` and `config.4,txt` files seem odd here anyway.

**Review by pvlugter — COMMENTED** [review]
> I wonder what the story is about that PR. Shouldn't it be made on the lightbend/lightbend-labs organization instead of here at the Cloudstate repository?

Yes. I expect that's the plan.

File: setup.py [hunk 5]
```diff
@@ -47,25 +54,26 @@ def run(self):
 setup(
     name=name,
     version=version,
-    url="https://github.com/cloudstateio/python-support",
+    url="https://github.com/jpollock/akkaserverless-python-sdk",
     license="Apache 2.0",
-    description="Cloudstate Python Support Library",
+    description="Akka Serverless Python Support Library",
```
Comment (marcellanz) [inline setup.py:59]:
```suggestion
    description="Akka Serverless Python SDK",
```

File: setup.cfg [hunk 6]
```diff
@@ -9,9 +9,11 @@ classifiers =
     Operating System :: OS Independent
     Programming Language :: Python
     Programming Language :: Python :: 3.6
+long_description = file: READ.md
```
Comment (marcellanz) [inline setup.cfg:12]:
```suggestion
long_description = file: README.md
```

File: setup.cfg [hunk 7]
```diff
@@ -9,9 +9,11 @@ classifiers =
...
+long_description_content_type = text/markdown
 project_urls =
-    Documentation = https://cloudstate.io/docs/user/lang/index.html
-    Source = https://github.com/cloudstateio/python-support
+    Documentation = https://akkaserverless.io/docs/user/lang/index.html
```
Comment (marcellanz) [inline setup.cfg:15]: There might be a reason to mix akkaserverless.io (here) and akkaserverless.com (at url) from time to time; or not.

File: README.md (hunk 3 above)
Comment (marcellanz) [inline README.md:8]:
```suggestion
This package is a fork of the original [User Language Support for Python](https://github.com/cloudstateio/python-support) for [Cloudstate](https://cloudstate.io/). This Akka Serverless package is heavily indebted to the [C]loudstate contributors](https://github.com/cloudstateio/python-support/graphs/contributors), especially [Adriano Santos](https://github.com/sleipnir) and [Marcel Lanz](https://github.com/marcellanz).
```

File: akkaserverless/akkaserverless_service.py [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    __port = "8080"
+    __workers = multiprocessing.cpu_count()
+    __event_sourced_entities: List[EventSourcedEntity] = field(default_factory=list)
+    __value_entities: List[ValueEntity] = field(default_factory=list)
+    __action_protocol_entities: List[Action] = field(default_factory=list)
+    __views: List[View] = field(default_factory=list)
+
+    def host(self, address: str):
+        """Set the address of the network Host.
+        Default Address is 127.0.0.1.
```
Comment (marcellanz) [inline akkaserverless/akkaserverless_service.py:50]: on line https://github.com/cloudstateio/python-support/pull/47/files#diff-296ae8741e7ca1d6af23a82352bc2fa1814a916d2841f3c30179550c52b1b4c9R40 it defaults to 0.0.0.0.

File: akkaserverless/akkaserverless_service.py [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    def port(self, port: str):
+        """Set the address of the network Port.
+        Default Port is 8080.
+        """
+        self.__port = port
+        return self
+
+    def max_workers(self, workers: Optional[int] = multiprocessing.cpu_count()):
+        """Set the gRPC Server number of Workers.
+        Default is equal to the number of CPU Cores in the machine.
```
Comment (marcellanz) [inline akkaserverless/akkaserverless_service.py:64]: In Serverless there is no machine anymore.

File: akkaserverless/akkaserverless_service.py [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+            self.__event_sourced_entities.append(component)
+        elif isinstance(component, ValueEntity):
+            self.__value_entities.append(component)
+        elif isinstance(component, View):
+            self.__views.append(component)
+        return self
+        
+
+    def start(self):
+        """Start the user function and gRPC Server."""
```
Comment (marcellanz) [inline akkaserverless/akkaserverless_service.py:82]: Don't you start `components` in AS, instead of `user functions`?

**Comment by sleipnir** [comment]
> Awesome @jpollock ! great to see contributions to the python user language support for Cloudstate.
> 
> With changes seen like
> `pip install -U akkaserverless`
> 
> I wonder what the story is about that PR. Shouldn't it be made on the lightbend/lightbend-labs organization instead of here at the Cloudstate repository? This PR transforms a user language support library for cloudstate into a Akka Serverless (AS) SDK compatible with the AS protocol of some of the later beta releases and is incompatible with this project here.
> 
> I think it would be better to fork/import this repository into one of these organizations instead of transforming it here. The Github UI lets you import an existing repository while keeping the git history.

I completely agree with Marcel.
It just shouldn't be here.

**Comment by sleipnir** [comment]
Thank you @jpollock  for the PR but we do not have a history discussed between LB and the CloudState community that allows them to be literally incorporated into Cloudstate, especially in terms of name and evolution of the protocol.
This should be a fork of the original and republished under the management of LB repositories.

We have an [issue](https://github.com/cloudstateio/cloudstate/issues/541) that treats under the issue of the evolution of the CloudState protocol but LB responded that at the moment has no interest in evolving with Cloudstate. In these terms we do not have, in my opinion, how to accept this PR fully here. This should be fork it under aegis of LB.

**Comment by jpollock** [comment]
Sorry! So so sorry!!!!

This was my bad. It was late at night and I intended to update my fork of this SDK: https://github.com/jpollock/akkaserverless-python-sdk.

I am beyond embarrassed that you spent time on this. I thought that I had checked the open PRs last night to confirm that my mistake was not what I thought it was but then woke this morning to see this comment chain. Ack. Again, my sincerest apologies for the confusion and wasted time.

**Comment by marcellanz** [comment]
@jpollock no worries.

>and wasted time.

not all was wasted, there are some real comments to be used :)

**Comment by marcellanz** [comment]
@jpollock Also, I knew you at LB have higher standards for a PR. So after reading the PR it came to me that this might be accidental.

**Pull request closed without merging**